modules/calib3d/src/p3p.cpp

```diff
@@ -100,39 +105,27 @@ bool p3p::solve(double R[3][3], double t[3],
 {
     double Rs[4][3][3], ts[4][3];
 
-    int n = solve(Rs, ts, mu0, mv0, X0, Y0, Z0,  mu1, mv1, X1, Y1, Z1, mu2, mv2, X2, Y2, Z2);
+    const bool p4p = true;
+    int n = solve(Rs, ts, mu0, mv0, X0, Y0, Z0,  mu1, mv1, X1, Y1, Z1, mu2, mv2, X2, Y2, Z2, mu3, mv3, X3, Y3, Z3, p4p);
 
     if (n == 0)
         return false;
 
-    int ns = 0;
```

Comment:
Should we ensure that `n == 1` here?

Reply:
The previous call to
`int n = solve(Rs, ts, mu0, mv0, X0, Y0, Z0,  mu1, mv1, X1, Y1, Z1, mu2, mv2, X2, Y2, Z2, mu3, mv3, X3, Y3, Z3, p4p);`
will return a maximum of 4 solutions.
And we choose the first one since the solution are sorted by reprojection error using the 4th point.

Reply:
OK, I see sorting code.